src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -5,10 +5,37 @@
 import core.basesyntax.model.User;
 
 public class RegistrationServiceImpl implements RegistrationService {
+    private static final int YEAR = 1;
+    private static final int MIN_AGE = 18;
+    private static final int MIN_CHAR_TO_CREATE_LOGIN_OR_PASSWORD = 6;
     private final StorageDao storageDao = new StorageDaoImpl();
 
     @Override
     public User register(User user) {
-        return null;
+        if (user.getLogin() == null) {
+            throw new RegistrationException("Needed login for registration process");
+        }
+        if (user.getPassword() == null) {
+            throw new RegistrationException("Needed password for registration process");
+        }
+        if (storageDao.get(user.getLogin()) != null) {
+            throw new RegistrationException("Try another one");
```

Comment:
Change the error message to a more informative one, e.g., 'User with this login already exists'